recipes/arcus/all/conanfile.py

```diff
@@ -0,0 +1,127 @@
+from conans import ConanFile, CMake, tools
+import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
+
+
+class ArcusConan(ConanFile):
+    name = "arcus"
+    description = "This library contains C++ code and Python3 bindings for " \
+                  "creating a socket in a thread and using this socket to send " \
+                  "and receive messages based on the Protocol Buffers library."
+    license = "LGPL-3.0-or-later"
+    topics = ("arcus", "protobuf", "socket", "cura")
+    homepage = "https://github.com/Ultimaker/libArcus"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("protobuf/3.17.1")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
```

Comment:
Could these be moved to patches?

Reply:
I don't know yet, we may have to live with these patches forever.